testutil/src/main/java/org/spine3/test/TimeTests.java

```diff
@@ -0,0 +1,180 @@
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package org.spine3.test;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.protobuf.Timestamp;
+import com.google.protobuf.util.Timestamps;
+import org.spine3.base.Command;
+import org.spine3.protobuf.Timestamps2;
+
+import static com.google.protobuf.util.Durations.fromSeconds;
+import static com.google.protobuf.util.Timestamps.add;
+import static com.google.protobuf.util.Timestamps.subtract;
+import static org.spine3.protobuf.Durations2.hours;
+import static org.spine3.protobuf.Timestamps2.getCurrentTime;
+import static org.spine3.protobuf.Timestamps2.systemTime;
+import static org.spine3.validate.Validate.checkPositive;
+
+/**
+ * Utility class for working with time-related tests.
+ *
+ * @author Alexander Yevsykov
+ */
+@VisibleForTesting
+public class TimeTests {
+
+    private TimeTests() {
+        // Prevent instantiation of this utility class.
+    }
+
+    /**
+     * Returns the {@linkplain Timestamps2#getCurrentTime() current time} in seconds.
+     *
+     * @return a seconds value
+     */
+    public static long currentTimeSeconds() {
+        final long secs = getCurrentTime().getSeconds();
+        return secs;
+    }
+
+    /**
+     * Adjusts a timestamp in the context of the passed command.
+     *
+     * @return new command instance with the modified timestamp
+     */
+    public static Command adjustTimestamp(Command command, Timestamp timestamp) {
+        final Command.Builder commandBuilder =
+                command.toBuilder()
+                       .setContext(command.getContext()
+                                          .toBuilder()
+                                          .setTimestamp(timestamp));
+        return commandBuilder.build();
+    }
+
+    /**
+     * The provider of current time, which is always the same.
+     *
+     * <p>Use this {@code Timestamps.Provider} in time-related tests that are
+     * sensitive to bounds of minutes, hours, days, etc.
+     */
+    public static class FrozenMadHatterParty implements Timestamps2.Provider {
+        private final Timestamp frozenTime;
+
+        public FrozenMadHatterParty(Timestamp frozenTime) {
+            this.frozenTime = frozenTime;
+        }
+
+        /** Returns the value passed to the constructor. */
+        @Override
+        public Timestamp getCurrentTime() {
+            return frozenTime;
+        }
+    }
+
+    /**
+     * The time provider that can rewind current time.
+     *
+     * <p>Created in the future, {@linkplain #THIRTY_YEARS_IN_HOURS 30 years} from
+     * the {@link Timestamps2#systemTime() current system time}.
+     */
+    public static class BackToTheFuture implements Timestamps2.Provider {
```

Comment:
:) I don't think we violate copyright here. I selected such a name deliberately because:
1. It prevents getting such a class into production.
2. The class behaviour is expected from the movie.